doc/content/design/trusted-certificates.md

```diff
@@ -0,0 +1,140 @@
+---
+title: Trusted certificates for identity validation in TLS connections
+layout: default
+design_doc: true
+revision: 1
+status: draft
+---
+
+# Overview
+
+In various use cases, TLS connections are established on the host on which XAPI runs.
+When establishing a TLS connection, the peer identity needs to be validated.
+This is done using either a root CA certificate to perform certificate chain validation, or a known peer certificate for validation with certificate pinning.
+The root CA certificates and peer certificates involved in this process are referred to as trusted certificates.
+When a trusted certificate is installed, the local endpoint can validate the peer identity during TLS connection establishment.
+Certificate chain validation is a general-purpose, standards-based approach but requires additional steps, such as getting the peer's certificate signed by a CA.
+In contrast, certificate pinning offers a quicker way to set up trust in some cases without the overhead of CA signing.
+For example, in Trust‑On‑First‑Use (TOFU) policy and a self-signed server certificate scenario the first received leaf certificate can be trusted to avoid user intervention or configuration.
```

Comment:
This reduces security as well.

If the service the hosts connects to is going to be known ahead of time (like in the case of licensing) A certificates can be added to the trust root using a package similar to `ca-certificates` to make it always available.

If it doesn't but the service is aware that it needs to be configured, the installation of the peer certificate can be automated using the API, this is the case for WLB.

Are there any more cases that need to be covered? I strongly suggest to stay away from TOFU in this case, and find alternative solutions that do not reduce security.